server/tweetSearch.js

```diff
@@ -1,7 +1,40 @@
 module.exports = function(client) {
     var tweetStore = [];
+    var tweetUpdates = [];
     var hashtags = ["#bristech", "#bristech2016"];
 
+    function addTweetItem(tweets, tag) {
+        tweetUpdates.push({
+            type: "new_tweets",
+            since: new Date(),
+            tag: tag,
+            startIdx: tweetStore.length,
+        });
+        tweetStore = tweetStore.concat(tweets);
+    }
+
+    function deleteTweet(tweetId) {
+        // Not really a necessary check, and can be taken out if the performance hit becomes too large, but acts as a
+        // way of preventing deletes of tweets that haven't appeared yet.
+        // This is important, as such deletes would result in tweets disappearing from queries as the "since" time is
+        // moved back - which would be an unintuitive and confusing behaviour.
+        var deletedTweet = tweetStore.find(function(tweet) {
+            return tweet.id_str === tweetId;
+        });
+        if (!deletedTweet) {
+            throw new Error("Cannot delete tweet that the server does not have.");
```

Comment:
We should probably catch this somewhere? I think the post request that will call this just returns 200 no matter what.